data/block/blockV2.go

```diff
@@ -629,3 +630,25 @@ func (hv2 *HeaderV2) GetAdditionalData() headerVersionData.HeaderAdditionalData
 	}
 	return additionalVersionData
 }
+
+// CheckFieldsForNil checks a predefined set of fields for nil values
+func (hv2 *HeaderV2) CheckFieldsForNil() error {
+	if hv2 == nil {
+		return data.ErrNilPointerReceiver
+	}
+	err := hv2.Header.CheckFieldsForNil()
+	if err != nil {
+		return err
+	}
+	if hv2.ScheduledRootHash == nil {
```

Comment:
Can Scheduled roothash be nil? Or it is equal with roothash? On sovereign this is certainly nil

Reply:
will remove this check, it is handled in the scheduled processor